### DISCUSSION TITLE
Gatsby Store using only 1 Collection

### DISCUSSION THREAD

**Discussion by dancrodev** [GatsbyStorefront/gatsby-theme-storefront-shopify#20] [Q&A] [answered]
Hi guys!

Not sure if this is conceptual or a matter of lack of knowledge, but most repos I've attempted for Gatsby Headless Storefronts, is that my 10,000+ variant store will prevent the build b/c of timeouts, etc. Is there a way to modify his repo to retrieve only 1 collection (images, products, etc.) and make a headless store completely from 1 shopify collection vs. pulling all products/images in?

I'm not looking for you to do the work, I just need pointed in the correct direction if it's possible. 

Thank you.

**Comment by paveli** [comment, marked as answer]
Hi @dancrodev, 
Surely this is possible. One way to achieve it is to fork `gatsby-source-shopify` plugin and modify it to fetch only one collection that you are interested in (or only products with some tag in it). Then use your modified plugin to fetch data from Shopify with Gatsby Storefront.

But first I would suggest trying to play with `GATSBY_STALL_RETRY_LIMIT`, `GATSBY_STALL_TIMEOUT`, `GATSBY_CONNECTION_TIMEOUT`, `GATSBY_INCOMPLETE_RETRY_LIMIT` to eliminate timeouts.

**Comment by paveli** [comment]
@dancrodev also you may take a look at this package - https://github.com/gatsbyjs/gatsby-source-shopify-experimental
This is an experimental approach to use Shopify's bulk operations API for data fetching. 
Though this plugin is not compatible with Gatsby Storefront yet, it may help in case you are developing your storefront from scratch.
